### ISSUE TITLE
INTERFACE for kvlds ADD: status is reversed

### ISSUE THREAD

**Issue by gperciva** [Tarsnap/kivaloo#89]
According to `INTERFACE`,

```
ADD:    Request type = 0x00000112
...
        Response (value set):
        [4 byte status code = 0]

        Response (value not set):
        [4 byte status code = 1]
```

However, this seems to be the opposite of what actually happens.  `kvlds/dispatch_mr.c`, on line 419, sets:
```
                /* Record if we did something. */
                if (op != OP_NONE)
                        req->opdone = 1;
```

Then, on line 513,
```
                case PROTO_KVLDS_ADD:
                        if (proto_kvlds_response_add(B->WQ, R->ID,
                            req->opdone))
                                goto err0;
                        break;
```

where `proto_kvlds_response_add()` has been defined in `lib/proto_kvlds_proto_kvlds.h` as:
```
#define proto_kvlds_response_add(Q, ID, status) \
        proto_kvlds_response_status(Q, ID, status)
```

As a result, if we successfully add a key (say, starting from an `rm -rf`'d LBS server), we receive status 1.
(At first glance, this applies to `MODIFY` as well.)

I'm content change the code and update the tests to not fail [*], but I'm wondering if you actually want another status code.  Namely (with extra verbosity),
- 0: value was set as a result of this ADD
- 1: value was not set because an error occurred
- 2: value was not set because there was already a key of this name

[*] at the moment, the only two tests for kvlds `ADD` are no-ops.  One of those is explicitly marked as a no-op, but the other lacks a `(should be a no-op)` tag.

**Comment by cperciva** [comment]
The code should match the documentation.  In all of the conditional operations, a status of `0` should mean that the precondition passed and the operation took place.

Please write tests and fix any bugs.

There's no need for "an error occurred" because if kvlds fails it dies.

**Comment by gperciva** [comment]
Fixed in #101.
